pkg/kubelet/kubelet_pods_test.go

```diff
@@ -834,20 +834,22 @@ func TestMakeEnvironmentVariables(t *testing.T) {
 	}
 
 	testCases := []struct {
-		name            string                 // the name of the test case
-		ns              string                 // the namespace to generate environment for
-		container       *v1.Container          // the container to use
-		masterServiceNs string                 // the namespace to read master service info from
-		nilLister       bool                   // whether the lister should be nil
-		configMap       *v1.ConfigMap          // an optional ConfigMap to pull from
-		secret          *v1.Secret             // an optional Secret to pull from
-		expectedEnvs    []kubecontainer.EnvVar // a set of expected environment vars
-		expectedError   bool                   // does the test fail
-		expectedEvent   string                 // does the test emit an event
+		name               string                 // the name of the test case
+		ns                 string                 // the namespace to generate environment for
+		enableServiceLinks bool                   // enabling service links
+		container          *v1.Container          // the container to use
+		masterServiceNs    string                 // the namespace to read master service info from
+		nilLister          bool                   // whether the lister should be nil
+		configMap          *v1.ConfigMap          // an optional ConfigMap to pull from
+		secret             *v1.Secret             // an optional Secret to pull from
+		expectedEnvs       []kubecontainer.EnvVar // a set of expected environment vars
+		expectedError      bool                   // does the test fail
+		expectedEvent      string                 // does the test emit an event
 	}{
 		{
-			name: "api server = Y, kubelet = Y",
-			ns:   "test1",
+			name:               "api server = Y, kubelet = Y",
+			ns:                 "test1",
+			enableServiceLinks: false,
```

Comment:
It's not clear what these test cases think they are testing, but you're changing it.  Since the default is true, you should probably handle this as:

1) add `enableServiceLinks: true` to each test case
2) add new test cases (with comments, please) that explicitly exercise the `false` case of that and nothing else.

If you feel generous you could refactor this test, making a util function that takes a list of test cases and a bunch of smaller Test* functions that call that util function with more targetted cases, e.g `TestMakeEnvironmentVariablesWithExpansions` and `TestMakeEnvironmentVariablesFromConfigMap` etc.  You can do that a a pre-factoring commit in this same PR.  If you feel generous...  It would make this testing much nicer :)